### PR TITLE
add pb-path flag

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

**Review by tailingchen — APPROVED** [review]

File: cmd/grpc-contract/main.go [hunk 1]
```diff
@@ -91,46 +91,73 @@ func main() {
...
+			[]string{
+				file,
+				path.Join(pbPath, goType+".pb.go"),
+				path.Join(pbPath, "messages.pb.go"),
+			})
+
+		// find the corresponding server interface
+		f, ok := pbFilesMap[contract.Sources[1]]
+		if !ok {
+			fmt.Printf("Failed to corresponding source file for service %v\n", goType)
```
Comment (yutelin) [inline cmd/grpc-contract/main.go]: Failed to "load" corresponding source files?

File: cmd/grpc-contract/main.go [hunk 2]
```diff
@@ -91,46 +91,73 @@ func main() {
...
+		}
+		var serverInterface *parser.GoInterface
+		for _, i := range f.Interfaces {
+			if contract.IsServerInterface(i.Name) {
+				serverInterface = i
+				break
+			}
+		}
+		if serverInterface == nil {
+			fmt.Printf("Failed to corresponding server interface for service %v\n", goType)
```
Comment (yutelin) [inline cmd/grpc-contract/main.go]: ditto

File: cmd/grpc-contract/main.go [hunk 3]
```diff
@@ -91,46 +91,73 @@ func main() {
...
+			os.Exit(-1)
+		}
+
+		// find the corresponding server interface
+		f, ok = pbFilesMap[contract.Sources[2]]
+		if !ok {
+			fmt.Printf("Failed to messages source file for service %v\n", goType)
```
Comment (yutelin) [inline cmd/grpc-contract/main.go]: Failed to find corresponding server interface?

File: cmd/grpc-contract/main.go [hunk 4]
```diff
@@ -91,46 +91,73 @@ func main() {
...
-
-					contract.Methods = append(contract.Methods, impl.NewMethod(pbPackage, m, request, response, goBindingFile, contract.StructName))
-				}
-				break
+		for _, m := range serverInterface.Methods {
+			// Find request struct
+			requestStructName := m.Params[1].Type[1:]
+			request := findGoStruct(requestStructName, f)
+			if request == nil {
+				fmt.Printf("Failed to corresponding request struct in method %v\n", m.Name)
```
Comment (yutelin) [inline cmd/grpc-contract/main.go]: ditto

File: cmd/grpc-contract/main.go [hunk 5]
```diff
@@ -91,46 +91,73 @@ func main() {
...
+				os.Exit(-1)
+			}
+
+			// Find response struct
+			responseStructName := m.Results[0].Type[1:]
+			response := findGoStruct(responseStructName, f)
+			if response == nil {
+				fmt.Printf("Failed to corresponding response struct in method %v\n", m.Name)
```
Comment (yutelin) [inline cmd/grpc-contract/main.go]: ditto

**Review by yutelin — APPROVED** [review]

File: cmd/grpc-contract/main.go (hunk 1 above)
Comment (markya0616) [reply cmd/grpc-contract/main.go]: OK

File: cmd/grpc-contract/main.go (hunk 2 above)
Comment (markya0616) [reply cmd/grpc-contract/main.go]: OK

File: cmd/grpc-contract/main.go (hunk 3 above)
Comment (markya0616) [reply cmd/grpc-contract/main.go]: OK

File: cmd/grpc-contract/main.go (hunk 4 above)
Comment (markya0616) [reply cmd/grpc-contract/main.go]: OK

File: cmd/grpc-contract/main.go (hunk 5 above)
Comment (markya0616) [reply cmd/grpc-contract/main.go]: OK

**Pull request merged**